ui/app/components/app/dropdowns/network-dropdown.js

```diff
@@ -125,6 +133,7 @@ class NetworkDropdown extends Component {
           closeMenu={() => this.props.hideNetworkDropdown()}
           onClick={() => {
             if (isPrefixedFormattedHexString(chainId)) {
+              setPreviousProvider(providerType)
```

Comment:
What does this do when `providerType` is `rpc` :thinking: 

Maybe this is a separate bug that is out of scope here, but I think it would blow up in that case, as `setProviderType` in the network controller asserts that `type` isn't `rpc`.

Reply:
Hmm, good point. I'll address this in a follow-up.